code/modules/spells/spell_types/rot_spells.dm

```diff
@@ -0,0 +1,132 @@
+
+
+/obj/effect/proc_holder/spell/aimed/canker
+	name = "Rotten Invocation: Canker"
+	desc = "This spell fires a bolt of rot at a target. While this spell may have once been great magicks, all it is now is mental sludge."
+	school = "evocation"
+	charge_max = 60
+	clothes_req = FALSE
+	invocation = "BUBOES, PHLEGM, BLOOD AND GUTS"
+	invocation_type = "shout"
+	range = 20
+	cooldown_min = 20 //10 deciseconds reduction per rank
+	projectile_type = /obj/item/projectile/magic/rot
+	base_icon_state = "rotten"
+	sound = 'sound/magic/fireball.ogg' //todo: replace with something better.
+	active_msg = "You prepare to cast a bolt of rot!"
+	deactive_msg = "You rinse your Invocation... for now."
+	active = FALSE
+	rotten_spell = TRUE
+
+	action_icon_state = "rotten0"
+	action_background_icon_state = "bg_rotting"
+
+/obj/effect/proc_holder/spell/targeted/forcewall/rotwall
+	name = "Rotten Invocation: Wall of Pestilence"
+	desc = "Create a rotten barrier that only you can pass through without decaying."
+	invocation = "BASK IN FILTH"
+	clothes_req = FALSE
+	sound = 'sound/magic/forcewall.ogg'//it's own sound todo
+	action_icon_state = "rotwall"
+	action_background_icon_state = "bg_rotting"
+	wall_type = /obj/effect/forcefield/rotwall
+	rotten_spell = TRUE
+
+/obj/effect/forcefield/rotwall
+	name = "Rotten Flesh"
+	desc = "A barrier of pestilence. You feel as if you could go through, but dark energies emanate from it..."
+	icon_state = "rotwall"
+	var/mob/wizard
+
+/obj/effect/forcefield/rotwall/Initialize(mapload, mob/summoner)
+	. = ..()
+	wizard = summoner
+	var/turf/T = get_turf(src.loc)
+	if(T)
+		var/datum/gas_mixture/stank = new
+		ADD_GAS(/datum/gas/miasma, stank.gases)
+		stank.gases[/datum/gas/miasma][MOLES] = 100 //300 moles in total
+		T.assume_air(stank)
+		T.air_update_turf()
```

Comment:
This is duplicated twice, put it in it's own proc

Reply:
Which file would be best to do that?

Reply:
Which file would be best to do that?

Reply:
actually, i'm fairly sure atmos_spawn_air is a proc that does exactly what I want for me, i'll just replace these with that.